### PR TITLE
Renames `WeightMeter` constructors for better readability

### PR DESCRIPTION
**Pull request by gupnik**
As discussed in the Elements chat, this PR adds the following changes:
1. Renames `WeightMeter::max_limit()` to `WeightMeter::new()` as the maximum weight limit is implied
2. Renames `WeightMeter::from_limit()` to `WeightMeter::with_limit()`

### CODE REVIEW

**Review by ggwpez — APPROVED** [review]

File: substrate/primitives/weights/src/weight_meter.rs
Comment (bkchr) [inline substrate/primitives/weights/src/weight_meter.rs:59]: But the old one was more expressive?

Comment (gupnik) [reply substrate/primitives/weights/src/weight_meter.rs:59]: This comes after a discussion with @gavofyork around the readability of this piece of code:
```rust
let meter = WeightMeter::max_limit();
meter.consume();
```
It reads as though the meter starts from the limit and should saturate with the further `consume` call. Going through the impl makes it clear that it still starts from 0.

Since the limit to max is implied with a meter, we should not need to specify it explicitly.

Comment (bkchr) [reply substrate/primitives/weights/src/weight_meter.rs:59]: Maybe `with_max_limit`? I mean your example is understandable, but `new` is still not really expressive to what the limit will be. (I will also not block on this, just some thought)

**Review by bkchr — APPROVED** [review]

**Review by ordian — APPROVED** [review]
(approve to unblock the merge)

**Pull request merged**
